lib/Parse/ParseType.cpp

```diff
@@ -1555,6 +1555,8 @@ bool Parser::canParseType() {
     consumeToken();
   } else if (Tok.isContextualKeyword("each")) {
     consumeToken();
+  } else if (Tok.isContextualKeyword("sending")) {
+    consumeToken();
```

Comment:
Is this correct? My impression is that only parameters can be annotated as `sending` and not arbitrary types. Looking at this change, I think it might cause the following to stop compiling because `sending` now gets consumed as a contextual keyword instead of as the type name.

```swift
struct sending {}
let a: sending = sending()
func test(x: sending) {}
```

Reply:
@ahoppen I just tested this locally and it isn't an issue.

Reply:
I just attached a debugger to it and understand why my example didn’t hit the issue that I was thinking about but this one does

```swift
struct sending {}
func test() {
  _ = { () -> sending in
  }
}
```

I think the proper solution here is to check if `sending` is followed by something that satisfies `canParseType`. If not, we probably need to backtrack and check if `sending` itself can be used as the type name.